src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -418,4 +418,236 @@ public static JointGridContainer combineToJointGrid(
+                    // node A is part of this subgrid -> no slack needed
+                    // internal node is needed for node admittance matrix and will be added
+
+                    // add the same transformer again to the new transformer set as nothing has
+                    // changed for this transformer
+                    resTrafo3wToInternal =
+                        new AbstractMap.SimpleEntry<>(oldTrafo3w, oldTrafo3w.getNodeInternal());
+
+                  } else {
+                    // node B or C is part of this subgrid -> internal node becomes the new slack
+
+                    // if node A is marked as slack, unmark it as slack
+                    NodeInput newNodeA =
+                        oldTrafo3w.getNodeA().isSlack()
+                            ? copyNode(oldTrafo3w.getNodeA(), false)
+                            : oldTrafo3w.getNodeA();
+
+                    // we need to take care for this node in our node sets afterwards
+                    // (needs to be replaced by the new nodeA which might have been a slack before)
+                    oldToNewTrafo3WNodes.put(oldTrafo3w.getNodeA(), newNodeA);
+
+                    // create an update version of this transformer with internal node as slack and
+                    // add it to the newTrafos3w set
+                    Transformer3WInput newTrafo3w =
+                        copyTrafo3WWithInternalSlack(oldTrafo3w, newNodeA);
+
+                    // add the slack
+                    resTrafo3wToInternal =
+                        new AbstractMap.SimpleEntry<>(newTrafo3w, newTrafo3w.getNodeInternal());
+                  }
+                  return resTrafo3wToInternal;
+                })
+            .collect(
+                Collectors.toMap(
+                    AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+    // get old transformer2w high voltage nodes (nodeA)
+    Map<NodeInput, NodeInput> oldToNewTrafo2WNodes =
+        subGridContainer.getRawGrid().getTransformer2Ws().stream()
+            .map(
+                oldTrafo2w -> {
+                  NodeInput oldNodeA = oldTrafo2w.getNodeA();
+                  NodeInput newNodeA = copyNode(oldNodeA, true);
+
+                  return new AbstractMap.SimpleEntry<>(oldNodeA, newNodeA);
+                })
+            .distinct()
+            .collect(
+                Collectors.toMap(
+                    AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+    // build the updated 2w transformer with nodeA as slack and add it to our new set
+    Set<Transformer2WInput> newTrafos2w =
+        subGridContainer.getRawGrid().getTransformer2Ws().stream()
+            .map(
+                oldTrafo2w ->
+                    copyTrafo2WUpdateNodeA(
+                        oldTrafo2w, oldToNewTrafo2WNodes.get(oldTrafo2w.getNodeA())))
+            .collect(Collectors.toSet());
+
+    // update node input graphics (2 winding transformers only)
```

Comment:
Wouldn't it also affect the nodes "A", of a three winding transformer, that are set to `slack = false`?

Reply:
yep but I thought as we never drawn 3 winding transformers yet, there is no need to adapt this here - added the corresponding code though